### DISCUSSION TITLE
Select Statements in Simple Triggers Don't Seem to Work on API

### DISCUSSION THREAD

**Discussion by BigDarkAkula** [supabase/supabase#12819] [Questions] [answered]
``Following` this discussion on Stack Overflow:

https://stackoverflow.com/questions/75642686/postgres-triggered-function-returns-null-result-for-subqueries

Can someone tell me why this triggers pulls the data from the select statement when I add a new record in console but if I make the insert on API, the value comes back as NULL?

**********************************8
CREATE OR REPLACE FUNCTION public.api_request_sf_trigger_func()
    RETURNS trigger
    LANGUAGE 'plpgsql'
    COST 100
    VOLATILE NOT LEAKPROOF
AS $BODY$
  
Declare

vapi_key text;

BEGIN

vapi_key=(select api_key from api_info limit 1);

RAISE LOG '3rd Trigger vapi_key is %',vapi_key;

Return New;

END;
$BODY$;

**Comment by GaryAustin1** [comment, marked as answer]
Because the user doing the insert from the API does not meet RLS for your select table.  They either need to meet RLS, or you need to make the trigger a "security definer" function so it runs as the owner of the function, usually Postgres role, which does bypass RLS.

**Reply by BigDarkAkula** [reply]
You have it exactly right!  It was a damn permission issue and I spent my time trying to recode it.  Thanks for the help!
